### ISSUE TITLE
unexpected behavior of lazy objects whose expression is the name of an object that exists in the search path

### ISSUE THREAD

**Issue by hafen** [hadley/lazyeval#25]
```
# these work as expected
eval(substitute(log(mpg)), mtcars)
eval(substitute(mpg), mtcars)

# these do too
library(lazyeval)
lazy_eval(lazy(log(mpg)), mtcars)
lazy_eval(lazy(mpg), mtcars)

library(ggplot2) # has a data set "mpg"
lazy_eval(lazy(log(mpg)), mtcars)
# this one doesn't
lazy_eval(lazy(mpg), mtcars)
```

**Comment by gergness** [comment]
In your specific example, `lazy_eval(lazy(mpg, .follow_symbols = F), mtcars)` works, but #39 might be related if you're doing this within a function.

**Comment by hadley** [comment]
Fixed by #18
